recipes-bsp/secure-boot-otp-provisioning/README.md

```diff
@@ -0,0 +1,154 @@
+# IOT2050 secure boot customer key provisioning
+
+IOT2050 Advanced PG2 enables the SoC extend OTP area for storing the customer
+key used for signing the firmware & software that validated by SEBoot and afterwards.
+
+Below are the steps to programming keys into this OTP area.
+
+## Create the keys
+
+First, you need to generate at least two sets of key pair under the folder
+`files/keys`. At most you can have three key pair sets. An example of key pairs
+are already there for reference.
+
+These keys are named `MPK`, `SMPK` and `BMPK` respectively.
+
+- `MPK`: the first key for signing customized firmware and image.
+- `SMPK`: the second key for signing customized firmware and image. If unfortunately
+the first key is leaked, user can switch to use this key.
+- `BMPK`: the third key for signing customized firmware and image. If unfortunately
+the second keys is leaked as well, user can switch to use this key.
+
+```shell
+openssl req -x509 -newkey rsa:4096 -keyout custMpk.pem -nodes -outform pem -out custMpk.crt -sha256
+openssl req -x509 -newkey rsa:4096 -keyout custSmpk.pem -nodes -outform pem -out custSmpk.crt -sha256
+# optional the third key
+#openssl req -x509 -newkey rsa:4096 -keyout custBmpk.pem -nodes -outform pem -out custBmpk.crt -sha256
+```
+
+> :warning:
+> Do remove the example dummy keys under the keys folder!
+
+## OTP programming
+
+The OTP programming data is generated by running the script `files/make-otpcmd.sh`.
+The generated `otpcmd.bin` should be programmed into the flash, and during the
+next booting, the first stage bootloader (SEBoot) will detect it and will try to
+program it into the OTP area.
+
+To ease the key provisioning tasks, we provided an example firmware image in
+this repo to build an image containing both:
+
+- the otpcmd.bin
+- signed firmware
+
+so that the first booting of signed firmware and OTP programming could be combined
+together.
+
+You can make your own implement based on this blueprint, or you can use the script
+directly.
```

Comment:
One piece description I'm missing in this context: When the user ships a new version of the firmware, signed with the active key, OTP will can be updated to prevent rollbacks. How is that managed?

Reply:
[REVISION in iot2050-sign-fw.sh](https://github.com/siemens/meta-iot2050/blob/964487f895e9f738051d0c3495de6529ce745fad/recipes-bsp/u-boot/files/0023-iot2050-Add-script-for-signing-artifacts.patch#L93)

This one need be documented. Also need be modeled to u-boot rules. Will submitted a patch later in this PR.